src/mediapipe_internal/mediapipegraphdefinition.cpp

```diff
@@ -438,22 +438,38 @@ std::pair<std::string, mediapipe_packet_type_enum> MediapipeGraphDefinition::get
     return {"", mediapipe_packet_type_enum::UNKNOWN};
 }
 
+struct PythonResourceInitializationCleaningGuard {
+    bool shouldCleanup{true};
+    std::unordered_map<std::string, std::shared_ptr<PythonNodeResource>>& resource;
+    PythonResourceInitializationCleaningGuard(std::unordered_map<std::string, std::shared_ptr<PythonNodeResource>>& resource) :
+        resource(resource) {}
+    ~PythonResourceInitializationCleaningGuard() {
+        if (shouldCleanup) {
+            resource.clear();
+        }
+    }
+    void markSuccess() {
+        shouldCleanup = false;
+    }
+};
+
 Status MediapipeGraphDefinition::initializeNodes() {
 #if (PYTHON_DISABLE == 0)
+    PythonResourceInitializationCleaningGuard initializationCleaningGuard(this->pythonNodeResources);
```

Comment:
pythonResourcesCleaningGuard?
more directly says what the cleanup is about

Reply:
done